### ISSUE TITLE
Uploads failing using aws-cli cp from cron

### ISSUE THREAD

**Issue by gjanz** [aws/aws-cli#1411]
I'm attempting to use aws-cli cp to upload sql backups from a virtual machine to s3. However for three nights running, I've had errors. These errors never occur when I run these scripts from a shell myself.

I initially thought it was similar to #454 but the 100 headers are there. I can't work out why it's failing; there's a 200 header after the error is thrown, which I assume is s3 giving an OK.

Debug log is at https://gist.github.com/gjanz/8fe19d3770d53bce26a3

**Comment by mtdowling** [comment]
I see that there is an MD5 mismatch between what was sent and what we expected to send. This error was then retried several times before the process failed. We are validating that the returned ETag matches the MD5 of the file that was sent over the wire. If S3 returns a different ETag than the file MD5, then this exception is raised.

Can you show the command that you're executing manually that works?

**Comment by mtdowling** [comment]
Actually, the fix to this issue is simply updating to the latest version of the CLI. Some of the code related to ETag validation of responses was removed as it's not the best place to perform this validation.

I'll go ahead and close this issue, but let us know if it doesn't fix it and we'll reopen it.

**Comment by gjanz** [comment]
So it seems this has still been happening for me. More often than I realised. So yesterday I updated the aws-cli tools to check I was running something recent (it updated to 1.10.65), and checking last nights backups, three of the four failed - with only partial uploads. The successful fourth was only 1270541 bytes, so didn't use a multi-part upload.

Log is at https://gist.github.com/gjanz/115adbf6f126a66e20d9f1e25ff46bff

However it still completes without a problem if I manually run the script from an ssh shell.
